.github/workflows/ci.yml

```
@@ -17,13 +17,20 @@ jobs:
           - rails6.1
           - rails7.0
           - rails7.1
+          - rails7.2
+          - rails8.0
           - rails_main
+        exclude:
+          - ruby-version: '3.1'
+            gemfile: rails8.0
+          - ruby-version: '3.1'
+            gemfile: rails_main
```

I would drop Ruby 3.1 from line 13, and use `include` instead of `exclude`:
```suggestion
        include:
          - {ruby-version: '3.1', gemfile: rails6.1}
          - {ruby-version: '3.1', gemfile: rails7.0}
          - {ruby-version: '3.1', gemfile: rails7.1}
          - {ruby-version: '3.1', gemfile: rails7.2}
```
I find `include` makes it easier to visualize the matrix in my head.

---

I usually go with what means fewer lines, but I don’t have strong feelings on this. Replaced with your suggestion.